include/opw_kinematics/opw_kinematics_impl.h

```diff
@@ -116,17 +213,103 @@ Solutions<T> inverse(const Parameters<T>& params, const Transform<T>& pose) noex
   T theta5_vii = -theta5_iii;
   T theta5_viii = -theta5_iv;
 
-  T theta6_i = std::atan2(matrix(0, 1) * s23[0] * cos1[0] + matrix(1, 1) * s23[0] * sin1[0] + matrix(2, 1) * c23[0],
-                          -matrix(0, 0) * s23[0] * cos1[0] - matrix(1, 0) * s23[0] * sin1[0] - matrix(2, 0) * c23[0]);
-
-  T theta6_ii = std::atan2(matrix(0, 1) * s23[1] * cos1[1] + matrix(1, 1) * s23[1] * sin1[1] + matrix(2, 1) * c23[1],
-                           -matrix(0, 0) * s23[1] * cos1[1] - matrix(1, 0) * s23[1] * sin1[1] - matrix(2, 0) * c23[1]);
-
-  T theta6_iii = std::atan2(matrix(0, 1) * s23[2] * cos1[2] + matrix(1, 1) * s23[2] * sin1[2] + matrix(2, 1) * c23[2],
-                            -matrix(0, 0) * s23[2] * cos1[2] - matrix(1, 0) * s23[2] * sin1[2] - matrix(2, 0) * c23[2]);
+  T zero_threshold{ T(1e-6) };
```

Comment:
Hello I'm here to nitpick.

Double cast to T

Reply:
Done.